eduHub/src/main/kotlin/com/arsiu/eduhub/shared/streams/SharedAssignmentStream.kt

```diff
@@ -0,0 +1,39 @@
+package com.arsiu.eduhub.shared.streams
+
+import com.arsiu.eduhub.v2.assignmentsvc.output.reqreply.assignment.FindAllAssignmentStreamResponse
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Sinks
+
+/**
+ * This class is responsible for managing a shared assignment stream.
+ * It uses Reactor's Sink API to multicast assignment updates to multiple subscribers.
+ */
+@Component
+class SharedAssignmentStream {
+
+    private val logger: Logger = LoggerFactory.getLogger(SharedAssignmentStream::class.java)
+
+    // Sink that can hold multiple subscribers. It uses a backpressure buffer to handle situations where items
+    // are being produced faster than they can be consumed.
+    private val processor: Sinks.Many<FindAllAssignmentStreamResponse> = Sinks.many().multicast().onBackpressureBuffer()
+
+    // Expose the processor as a Flux so that multiple subscribers can listen for assignment updates.
+    val flux: Flux<FindAllAssignmentStreamResponse> = processor.asFlux().concatWith(Flux.never())
```

Comment:
why do you need .concatWith(Flux.never())?

Reply:
So shared flux won’t ended up empty.
If client open connection via grpc method that return stream and close it(So we don’t have any sub).
That will trigger onComplete method which will lead to processor.asFlux() to become empty.
But if we concat it with never flux won’t be terminated

Reply:
change it to sink.asFlux().publish().autoConnect(1) so now we have hot flux and basically covers edge case when sun stops listens to Flux

Reply:
you can have a shared sink but for each subscriber create a new flux